tests/migrations/test_update_legacy_filtered_positives.py

```diff
@@ -0,0 +1,193 @@
+from unittest.mock import patch
+import pandas as pd
+import pytest
+from crawler.filtered_positive_identifier import (
+    FilteredPositiveIdentifierV0,
+    FilteredPositiveIdentifierV1,
+    FilteredPositiveIdentifierV2,
+)
+
+from migrations import update_legacy_filtered_positives
+
+
+@pytest.fixture
+def mock_helper_database_updates():
+    with patch(
+        "migrations.update_legacy_filtered_positives.update_mongo_filtered_positive_fields"
+    ) as mock_update_mongo:
+        with patch(
+            "migrations.update_legacy_filtered_positives.update_mlwh_filtered_positive_fields"
+        ) as mock_update_mlwh:
+            yield mock_update_mongo, mock_update_mlwh
+
+
+@pytest.fixture
+def mock_user_input():
+    with patch("migrations.update_legacy_filtered_positives.get_input") as mock_user_input:
+        yield mock_user_input
+
+
+@pytest.fixture
+def mock_v0_version_set():
+    with patch("migrations.update_legacy_filtered_positives.v0_version_set") as mock_v0_version_set:
+        yield mock_v0_version_set
+
+
+@pytest.fixture
+def mock_query_helper_functions():
+    with patch("migrations.update_legacy_filtered_positives.legacy_mongo_samples") as mock_legacy_mongo_samples:
+        with patch(
+            "migrations.update_legacy_filtered_positives.get_cherrypicked_samples_by_date"
+        ) as mock_get_cherrypicked_samples_by_date:
+            yield mock_legacy_mongo_samples, mock_get_cherrypicked_samples_by_date
+
+
+@pytest.fixture
+def mock_extract_required_cp_info():
+    with patch("migrations.update_legacy_filtered_positives.extract_required_cp_info") as mock_extract_required_cp_info:
+        yield mock_extract_required_cp_info
+
+
+@pytest.fixture
+def mock_helper_functions():
+    with patch(
+        "migrations.update_legacy_filtered_positives.split_mongo_samples_by_version"
+    ) as mock_split_mongo_samples_by_version:
+        yield mock_split_mongo_samples_by_version
+
+
+@pytest.fixture
+def mock_update_filtered_positive_fields():
+    with patch(
+        "migrations.update_legacy_filtered_positives.update_filtered_positive_fields"
+    ) as mock_update_filtered_positive_fields:
+        yield mock_update_filtered_positive_fields
+
+
+# ----- test migration -----
+
+
+def test_update_legacy_filtered_positives_exception_raised_if_user_cancels_migration(
+    mock_user_input, mock_helper_database_updates, filtered_positive_testing_samples
+):
+    with pytest.raises(Exception):
+        mock_user_input.return_value = "no"
+        mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+
+        update_legacy_filtered_positives.run("crawler.config.integration")
+
+        mock_update_mongo.assert_not_called()
+        mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_exception_raised_if_user_enters_invalid_input(
```

Comment:
As in the migration file, think this and the first test should just check no database is updated and not raise an exception to skip the code

Reply:
I've changed these tests